### PR TITLE
Run tests on rails8

### PR DESCRIPTION
**Pull request by shmokmt**
## Description

Ruby on Rails 8.0 is out.
https://edgeguides.rubyonrails.org/8_0_release_notes.html

However, Rails8 requires Ruby 3.2 or higher version and latest jruby version is still Ruby 3.1 compatibility.
Considering these, I excluded fail jobs.
https://github.com/rails/rails/blob/5287cfa17681527761c50fd8c6ae169f4500bd97/rails.gemspec#L12
https://github.com/jruby/jruby/releases/tag/9.4.9.0

Thanks!

### CODE REVIEW

**Comment by shmokmt** [comment]
I checked that all tests passed.
https://github.com/shmokmt/asset_sync/pull/1

**Comment by PikachuEXE** [comment]
Please help me update all `3.0` to `"3.0"` to ensure it really test 3.0 (otherwise parsed as 3 = latest 3.x)
Also 3.0 need to be excluded

**Comment by shmokmt** [comment]
@PikachuEXE 
I fixed.
Could you review again?
https://github.com/AssetSync/asset_sync/pull/444/commits/f1f154582935a26608ad146f6d0224927f11d79e

All tests passed.
https://github.com/shmokmt/asset_sync/actions/runs/11964151388/job/33355968682

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -51,7 +65,7 @@ jobs:
       BUNDLE_GEMFILE: "${{ matrix.gemfile }}"
       ALLOW_FAILURES: "${{ matrix.allow_failures }}"
     runs-on: ${{ matrix.os }}-latest
-    continue-on-error: ${{ endsWith(matrix.ruby, 'head') || matrix.ruby == 'debug' }}
+    continue-on-error: ${{ endsWith(matrix.ruby, 'head') || matrix.ruby == 'debug' || matrix.ruby == 'jruby' }}
```
Comment (PikachuEXE) [inline .github/workflows/tests.yaml:68]: Is this needed? No test failure in jruby jobs

Comment (shmokmt) [reply .github/workflows/tests.yaml]: Oh, I forgot delete this condition. thanks!
I fixed.
https://github.com/AssetSync/asset_sync/pull/444/commits/a1710ddba21d367eb9174e28aff59b891ff8e9c5

Comment (shmokmt) [reply .github/workflows/tests.yaml]: @PikachuEXE

**Pull request merged**